src/main/main.cpp

```diff
@@ -2,22 +2,134 @@
+                    "current exception: NonSociRelatedException(\"%s\")\n",
+                    e.what());
+        }
+        catch (const CryptoError& e)
+        {
+            fprintf(stderr, "current exception: CryptoError(\"%s\")\n",
+                    e.what());
+        }
+        catch (const FileSystemException& e)
+        {
+            fprintf(stderr, "current exception: FileSystemException(\"%s\")\n",
+                    e.what());
+        }
+        catch (const InvariantDoesNotHold& e)
+        {
+            fprintf(stderr, "current exception: InvariantDoesNotHold(\"%s\")\n",
+                    e.what());
+        }
+        catch (const std::filesystem::filesystem_error& e)
+        {
+            fprintf(stderr,
+                    "current exception: std::filesystem::filesystem_error(%d, "
+                    "\"%s\", \"%s\", \"%s\", \"%s\")\n",
+                    e.code().value(), e.code().message().c_str(), e.what(),
+                    e.path1().c_str(), e.path2().c_str());
+        }
+        catch (const std::system_error& e)
+        {
+            fprintf(
+                stderr,
+                "current exception: std::system_error(%d, \"%s\", \"%s\")\n",
+                e.code().value(), e.code().message().c_str(), e.what());
+        }
+        catch (const std::logic_error& e)
+        {
+            fprintf(stderr, "current exception: std::logic_error(\"%s\")\n",
+                    e.what());
+        }
+        catch (const std::domain_error& e)
+        {
+            fprintf(stderr, "current exception: std::domain_error(\"%s\")\n",
+                    e.what());
+        }
+        catch (const std::invalid_argument& e)
+        {
+            fprintf(stderr,
+                    "current exception: std::invalid_argument(\"%s\")\n",
+                    e.what());
+        }
+        catch (const std::length_error& e)
+        {
+            fprintf(stderr, "current exception: std::length_error(\"%s\")\n",
+                    e.what());
+        }
+        catch (const std::out_of_range& e)
+        {
+            fprintf(stderr, "current exception: std::out_of_range(\"%s\")\n",
+                    e.what());
+        }
+        catch (const std::runtime_error& e)
+        {
+            fprintf(stderr, "current exception: std::runtime_error(\"%s\")\n",
+                    e.what());
+        }
+        catch (const std::range_error& e)
+        {
+            fprintf(stderr, "current exception: std::range_error(\"%s\")\n",
+                    e.what());
+        }
+        catch (const std::overflow_error& e)
+        {
+            fprintf(stderr, "current exception: std::overflow_error(\"%s\")\n",
+                    e.what());
+        }
+        catch (const std::underflow_error& e)
+        {
+            fprintf(stderr, "current exception: std::underflow_error(\"%s\")\n",
+                    e.what());
+        }
+        catch (const std::exception& e)
```

Comment:
we should remove the `try/catch` https://github.com/stellar/stellar-core/blob/d55d2bc98433c3440e3c5731049ecf31a1768d58/src/main/ApplicationUtils.cpp#L206-L210 as this ends up being redundant with this code (and by removing the `throw;` we actually capture the stack trace)

Reply:
fixed